packages/superset-ui-chart/src/components/NoResultsComponent.tsx

```diff
@@ -0,0 +1,35 @@
+import React, { CSSProperties } from 'react';
+
+const generateContainerStyles: (
+  height: number | string,
+  width: number | string,
+) => CSSProperties = (height: number | string, width: number | string) => ({
+  alignItems: 'center',
+  display: 'flex',
+  flexDirection: 'column',
+  height,
+  justifyContent: 'center',
+  padding: 16,
+  textAlign: 'center',
+  width,
+});
+
+type Props = {
+  height: number | string;
+  width: number | string;
+};
+
+const NoResultsComponent = ({ height, width }: Props) => (
+  <div style={generateContainerStyles(height, width)}>
```

Comment:
We can use `useMemo` hook for this to avoid creating new style object every time.

Reply:
done